### ISSUE TITLE
ReduceLROnPlateau does not work properly when choosing different monitor

### ISSUE THREAD

**Issue by danielcanueto** [keras-team/keras#11964]
Hi

I'm working from a Kaggle Notebook and I want to change the metric to monitor during ReduceLRonPlateau by another one (F1 score).

It seems the function is identifying properly the 'val_f1_metric' to monitor (look how the metric is reduced from inf to val_f1_metric after the first epoch) But, in the next epochs, the improvement of the metric is not identified:
 
![image](https://user-images.githubusercontent.com/21126465/50589453-f9270000-0e86-11e9-8d25-14c40679f45d.png)

I have the intuition taht, after the first epoch, the evaluation is being done on 'val_loss' instead of on 'val_f1_metric'.

The change of monitor is written like this:

reduce_lr = ReduceLROnPlateau(monitor='val_f1_metric', factor=0.5, patience=1, min_lr=0.0001, verbose=2)

Please let me know if there is something written wrong.

Thanks for the help

**Comment by danielcanueto** [comment]
Ok, I found the problem. I had to calculate 1 -f1 instead of f1. The issue can be closed.

**Comment by sallamander** [comment]
@danielcanueto you could also achieve the intended result of monitoring F1 (and not 1 - F1) by setting `mode=max` in the `ReduceLROnPlateau.__init__`. It's by default set to `auto`, which ends up acting as if `mode=min` since `acc` isn't in your metric name. Reference lines: 

https://github.com/keras-team/keras/blob/master/keras/callbacks.py#L1085

**Comment by gabrieldemarmiesse** [comment]
Closing as this can be solved with @sallamander 's solution.
